products/bridge/smart-contracts/contracts/periphery/BridgedToken.sol

```diff
@@ -29,7 +49,32 @@ contract BridgedToken is IERC20, ERC20, ERC20Burnable, Ownable {
         _mint(to, amount);
     }
 
+    function setLockProxyAddress(address lockProxyAddress_) external onlyOwner {
+        lockProxyAddress = lockProxyAddress_;
+    }
+
+    function transfer(
+        address to,
+        uint256 value
+    ) public override(ERC20, IERC20) returns (bool) {
+        mintIfLockProxy(msg.sender, to, value);
+        return super.transfer(to, value);
+    }
+
+    function transferFrom(
+        address from,
+        address to,
+        uint256 value
+    ) public override(ERC20, IERC20) returns (bool) {
+        mintIfLockProxy(msg.sender, to, value);
```

Comment:
You inherit from `ERC20` which inherits from `Context`, so you may want to use `_msgSender()` instead of `msg.sender`